api/workAllocation/util.ts

```diff
@@ -354,20 +354,24 @@ export async function getCaseIdListFromRoles(roleAssignmentList: RoleAssignment[
 
   let cases = [];
   caseResults.forEach( caseResult => cases = [...cases, ...caseResult.cases]);
+
   return cases;
 }
 
+export function filterMyAccessRoleAssignments(roleAssignmentList: RoleAssignment[]) {
+  return roleAssignmentList.filter(roleAssignment =>
+    (
+      roleAssignment.grantType === 'SPECIFIC' ||
+      roleAssignment.roleName === 'specific-access-requested' ||
+      roleAssignment.roleName === 'specific-access-denied'
+    ) &&
+    (!roleAssignment.attributes || roleAssignment.attributes.substantive !== 'Y')
+  );
```

Comment:
The predicate in this filter method is complex. Try and break it down. There really should be a maximum of 2-3 equality checks within a single statement for readability.